src/sagas/account.js

```diff
@@ -112,11 +118,18 @@ export function* loginFlow () {
 
     if (!(yield select(getUser))) {
       // Wait for login / signup if we don't have a user
-      const action = yield take([LOGIN, SIGNUP]);
-      if (action.type === LOGIN) {
-        yield call(doLogin, action);
-      } else {
-        yield call(doSignup, action);
+      const action = yield take([LOGIN, SIGNUP, SET_USER_TOKEN]);
+
+      switch (action.type) {
+        case LOGIN:
+          yield call(doLogin, action);
+          break;
+        case SIGNUP:
+          yield call(doSignup, action);
+          break;
+        case SET_USER_TOKEN:
+          yield call(loadUser);
```

Comment:
Does this need to pass the token?

Reply:
This is trigger by the `SET_USER_TOKEN` action so we already have the token